relay-general/src/protocol/replay.rs

```diff
@@ -0,0 +1,409 @@
+//! Replay processing and normalization module.
+//!
+//! Replays are multi-part values sent from Sentry integrations spanning arbitrary time-periods.
+//! They are ingested incrementally.
+//!
+//! # Protocol
+//!
+//! Relay is expecting a JSON object with some mandatory metadata.  However, environment and user
+//! metadata is usually sent in addition to the minimal payload.
+//!
+//! ```json
+//! {
+//!     "type": "replay_event",
+//!     "replay_id": "d2132d31b39445f1938d7e21b6bf0ec4",
+//!     "event_id": "63c5b0f895441a94340183c5f1e74cd4",
+//!     "segment_id": 0,
+//!     "timestamp": 1597976392.6542819,
+//!     "replay_start_timestamp": 1597976392.6542819,
+//!     "urls": ["https://sentry.io"],
+//!     "error_ids": ["d2132d31b39445f1938d7e21b6bf0ec4"],
+//!     "trace_ids": ["63c5b0f895441a94340183c5f1e74cd4"],
+//!     "request": {
+//!         "headers": {"User-Agent": "Mozilla/5.0..."}
+//!     },
+//! }
+//! ```
+use crate::protocol::{
+    ClientSdkInfo, Contexts, IpAddr, LenientString, Request, Tags, Timestamp, User,
+};
+use crate::store::is_valid_platform;
+use crate::store::user_agent::normalize_user_agent_generic;
+use crate::types::{Annotated, Array};
+use crate::user_agent;
+use std::net::IpAddr as RealIPAddr;
+
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+#[metastructure(process_func = "process_replay", value_type = "Replay")]
+pub struct Replay {
+    pub event_id: Annotated<String>,
+    pub replay_id: Annotated<String>,
```

Comment:
`replay_id` is typed as an `EventId` in other places. Should we use the same type here? https://github.com/getsentry/relay/blob/4cb923f0667da94801039ff2cbd99635221dd97b/relay-server/src/actors/store.rs#L569